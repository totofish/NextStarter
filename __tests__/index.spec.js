import { shallow, mount, render } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index'
import Layout from '../components/layout';

describe('Index With Enzyme', () => {
  const app = mount(<App />)
  it('Have text "== Index =="', () => {
    expect(app.html()).toContain('== Index ==');
  })

  it('Have one dom <nav>', () => {
    expect(app.find('nav')).toHaveLength(1);
  })
})

describe('Index Snapshot Testing', () => {
  it('Index Dom', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})