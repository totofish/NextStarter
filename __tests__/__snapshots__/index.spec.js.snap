// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Snapshot Testing Index Dom 1`] = `
<div>
  <header>
    <nav>
      <a
        href="/"
        onClick={[Function]}
      >
        Index
      </a>
      <br />
      <a
        href="/page/a"
        onClick={[Function]}
      >
        Page/a
      </a>
      <br />
      <a
        href="/page/b"
        onClick={[Function]}
      >
        Page/b
      </a>
      <br />
    </nav>
  </header>
  == Index ==
  <br />
  <br />
  <footer>
    Footer
  </footer>
</div>
`;
